### PR TITLE
Upgrade mranderson to 0.5.0

### PR DESCRIPTION
**Pull request by benedekfazekas**
mranderson 0.5.x specific config added to the project file as
well. **Resolved tree** mode is used. This means that
mranderson works with a resolved dependency tree and does not create a
deeply nested directory structure, only adds prefixes to all
dependencies. Handles all depedendencies as first level ones and
processes them in a topological order.

Add project prefix in the project file for mranderson, also some
config for **unresolved tree** if that would get used in the near
future.

Breaking changes in mranderson are followed up in Makefile and README

- [x] The commits are consistent with our [contribution guidelines](CONTRIBUTING.md)
- [ ] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [x] You've updated the README (if adding/changing middleware)

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -9,18 +9,19 @@
...
-                 ;; See https://github.com/clojure-emacs/cider/issues/2176 for details
-                 ^:source-dep [org.clojure/java.classpath "0.3.0"]
                  ^:source-dep [org.clojure/tools.namespace "0.3.0-alpha4"]
                  ^:source-dep [org.clojure/tools.trace "0.7.10"]
                  ^:source-dep [org.clojure/tools.reader "1.2.2"]]
-  :plugins [[thomasa/mranderson "0.4.9"]]
+  :plugins [[thomasa/mranderson "0.5.0"]]
+  :mranderson {:project-prefix "cider.inlined-deps"
+               :overrides       {[mvxcvi/puget fipp] [fipp "0.6.15"]}; only takes effect in unresolved-tree mode
+               :expositions     [[mvxcvi/puget fipp]]; only takes effect unresolved-tree mode
```
Comment (bbatsov) [inline project.clj]: Btw, shouldn't this be named `exposes` to be consistent with `overrides`. I guess it's consistent in a way even now, but I've never heard expositions used in this sense. :-)

File: project.clj [hunk 2]
```diff
@@ -9,18 +9,19 @@
...
-                 ;; Not used directly in cider-nrepl, but needed because of tools.namespace
-                 ;; and the way MrAnderson processes dependencies
-                 ;; See https://github.com/clojure-emacs/cider/issues/2176 for details
-                 ^:source-dep [org.clojure/java.classpath "0.3.0"]
                  ^:source-dep [org.clojure/tools.namespace "0.3.0-alpha4"]
                  ^:source-dep [org.clojure/tools.trace "0.7.10"]
                  ^:source-dep [org.clojure/tools.reader "1.2.2"]]
-  :plugins [[thomasa/mranderson "0.4.9"]]
+  :plugins [[thomasa/mranderson "0.5.0"]]
+  :mranderson {:project-prefix "cider.inlined-deps"
```
Comment (bbatsov) [inline project.clj]: I think this should be `cider.nrepl.inlined-deps`.

File: project.clj (hunk 1 above)
Comment (benedekfazekas) [reply project.clj]: as much as I googled it seemed that the noun form is `exposition`. eg something being exposed is an exposition (or an exposure but that has negative connotations).

Comment (bbatsov) [reply project.clj]: Well, I can live with the name. :-) My actually confusion was from the fact that `overrides` can be interpreted both as a verb and a noun and first I thought it was a verb. No a big deal.

File: project.clj [hunk 3]
```diff
@@ -9,18 +9,19 @@
...
-                 ^:source-dep [org.clojure/java.classpath "0.3.0"]
                  ^:source-dep [org.clojure/tools.namespace "0.3.0-alpha4"]
                  ^:source-dep [org.clojure/tools.trace "0.7.10"]
                  ^:source-dep [org.clojure/tools.reader "1.2.2"]]
-  :plugins [[thomasa/mranderson "0.4.9"]]
+  :plugins [[thomasa/mranderson "0.5.0"]]
+  :mranderson {:project-prefix "cider.nrepl.inlined-deps"
+               :overrides       {[mvxcvi/puget fipp] [fipp "0.6.15"]}; only takes effect in unresolved-tree mode
+               :expositions     [[mvxcvi/puget fipp]]; only takes effect unresolved-tree mode
+               :unresolved-tree false}
```
Comment (bbatsov) [inline project.clj]: Unrelated comment - why is this needed? Wasn't `:resolved-tree` the default anyways?

**Comment by bbatsov** [comment]
Thanks for all the work you've put in the new release!

File: project.clj (hunk 3 above)
Comment (benedekfazekas) [reply project.clj]: it is not really needed. but for now i left it like that with the above options that are not needed in resolved tree mode either to make it easier to switch to unresolved tree mode in the near future if the need arises. if not everything apart from the project prefix can be removed.

**Comment by benedekfazekas** [comment]
cheers :)

File: project.clj (hunk 3 above)
Comment (bbatsov) [reply project.clj]: My point was that I thought the default would be `:resolved-deps true` and its opposite would be `:resolved-tree false`. Might have been a bit better if this was `:mode :(un)resolved-tree`, but as long as it gets the job done anything goes.
